<launch>

    <!-- Configurable arguments -->
    <arg name="robot_name" default="sam"/>

    <arg name="loop_freq" default="20."/>
    <arg name="verbose_arg" default="True"/>

    <arg name="waypoint_topic_arg" default="/waypoint_topic"/>


    <!-- topics, note the root! -->
    <arg name="state_feedback_topic" default="/$(arg robot_name)/dr/odom" />
    <arg name="state_estimation_topic" default="/$(arg robot_name)/EstimatedPose" />
    <arg name="vbs_topic" default="/$(arg robot_name)/core/vbs_cmd" />
    <arg name="ref_pose_topic" default="$(arg waypoint_topic_arg)" />
    <arg name="dockingStationEstimationTopicArg" default="EstimatedStationPose"/>


    <!-- Nodes -->
    <node name="tf_broadcaster_DS" pkg="visual_feedback" type="tf_broadcaster.py"
        output="screen">

        <param name="frameName" value="ds_base_link"/>
        <param name="frameTopic" value="/sam/dr/ds_odom"/>
    </node>

    <node name="fakeTopics" pkg="visual_feedback" type="fakeTopicPublisher.py"
        output="screen">

        <param name="frameName" value="ds_base_link"/>
        <param name="frameTopic" value="/sam/dr/ds_odom"/>
    </node>

    <node name="tf_broadcaster_SAM" pkg="visual_feedback" type="tf_broadcaster.py"
        output="screen">

        <param name="frameName" value="sam/base_link"/>
        <param name="frameTopic" value="/sam/slam/odom"/>
    </node>


</launch>
